common/character_interactions/00_prison_interactions.txt

```diff
@@ -3546,7 +3546,26 @@ execute_prisoner_interaction = {
 				NOT = { has_character_flag = is_being_tortured }
 			}
 		}
-	}
+		# Warcraft
+		# Elunic Kaldorei AI will not execute Kaldorei Illidari prisoners # Hotfix until new criminal doctrines
+		trigger_if = {
+			limit = { 
+				scope:actor = { is_ai = yes }
```

Comment:
The AI behavior is set in `ai_will_do` block. So this entire feature should be moved there.